front/components/agent_builder/observability/shared/ChartShapes.tsx

```diff
@@ -28,10 +28,26 @@ export function RoundedTopBarShape({
   ) {
     return <g />;
   }
-  const r = 4;
-  if (!isTopForPayload(payload, seriesIdx)) {
+
+  const toolValue = payload.values[toolName] ?? 0;
+  if (toolValue <= 0) {
     return <rect x={x} y={y} width={width} height={height} fill={fill} />;
   }
+
+  let topTool: string | undefined;
+  for (let idx = stackOrder.length - 1; idx >= 0; idx--) {
+    const candidate = stackOrder[idx];
+    if ((payload.values[candidate] ?? 0) > 0) {
+      topTool = candidate;
+      break;
+    }
+  }
```

Comment:
Can we iterate over the items directly? instead of using the idx and then access the item